zebra-rpc/src/methods.rs

```diff
@@ -733,200 +746,212 @@ where
-                }
-            } else if verbosity == 1 || verbosity == 2 {
-                // # Performance
-                //
-                // This RPC is used in `lightwalletd`'s initial sync of 2 million blocks,
-                // so it needs to load all its fields very efficiently.
-                //
-                // Currently, we get the block hash and transaction IDs from indexes,
-                // which is much more efficient than loading all the block data,
-                // then hashing the block header and all the transactions.
-
-                // Get the block hash from the height -> hash index, if needed
-                //
-                // # Concurrency
-                //
-                // For consistency, this lookup must be performed first, then all the other
-                // lookups must be based on the hash.
-                //
-                // All possible responses are valid, even if the best chain changes. Clients
-                // must be able to handle chain forks, including a hash for a block that is
-                // later discovered to be on a side chain.
-
-                let should_read_block_header = verbosity == 2;
-
-                let hash = match hash_or_height {
-                    HashOrHeight::Hash(hash) => hash,
-                    HashOrHeight::Height(height) => {
-                        let request = zebra_state::ReadRequest::BestChainBlockHash(height);
-                        let response = state
-                            .ready()
-                            .and_then(|service| service.call(request))
-                            .await
-                            .map_server_error()?;
-
-                        match response {
-                            zebra_state::ReadResponse::BlockHash(Some(hash)) => hash,
-                            zebra_state::ReadResponse::BlockHash(None) => {
-                                return Err(Error {
-                                    code: MISSING_BLOCK_ERROR_CODE,
-                                    message: "block height not in best chain".to_string(),
-                                    data: None,
-                                })
-                            }
-                            _ => unreachable!("unmatched response to a block hash request"),
+            let hash_or_height: HashOrHeight = hash_or_height
+                .parse()
+                // Reference for the legacy error code:
+                // <https://github.com/zcash/zcash/blob/99ad6fdc3a549ab510422820eea5e5ce9f60a5fd/src/rpc/blockchain.cpp#L629>
+                .map_error(server::error::LegacyCode::InvalidParameter)?;
+
+            match verbosity.unwrap_or(1) {
+                0 => {
+                    // # Performance
+                    //
+                    // This RPC is used in `lightwalletd`'s initial sync of 2 million blocks,
+                    // so it needs to load block data very efficiently.
```

Comment:
I didn't find similar comments. Could you share which ones you had in mind?

Reply:
I can't apply this suggestion anymore, so I moved the note at the top of the fn in [2c342f7](https://github.com/ZcashFoundation/zebra/pull/9049/commits/2c342f7e1ad48e261ae6b32ebd423e83b34fe800). I moved it because the verbosity 1 arm is not in the code anymore.

Reply:
> I can't apply this suggestion anymore

It's essentially the intended change, I'd describe that as having applied the suggestion, though I'm not sure if that's accurate.

Reply:
I just meant that the "apply" button here on GitHub wasn't active anymore, so I did the suggested changes locally. The downside is that @oxarbitrage didn't get any credit in the commit history.